modules/avatar/react/lib/AvatarButton.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import styled from 'react-emotion';
+import isPropValid from '@emotion/is-prop-valid';
+import Avatar, {AvatarVariant} from './Avatar';
+import {colors} from '@workday/canvas-kit-react-core';
+import {focusRing, hideMouseFocus} from '@workday/canvas-kit-react-common';
+import {SystemIconCircleSize} from '@workday/canvas-kit-react-icon';
+
+export interface AvatarButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * An AvatarVariant enum indicating which variant to use for the default state (Light vs. Dark)
+   */
+  variant: AvatarVariant;
+  /**
+   * An SystemIconCircleSize enum or number value indicating the size of the avatar
+   */
+  size: SystemIconCircleSize | number;
+  /**
+   * Text describing what the avatar is showing
+   */
+  altText: string;
+  /**
+   * The url of the users avatar photo
+   */
+  url?: string;
+  /**
+   * An event handler function that gets called when the avatar is clicked
+   */
+  onClick?: (e: React.SyntheticEvent) => void;
+  /**
+   * Ref of button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+}
+
+const Container = styled('button', {
+  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size',
+})<Omit<AvatarButtonProps, 'altText'>>(
+  {
+    display: 'flex',
```

Comment:
None of these base styles are needed except for `borderRadius`. They are all handled by the internal `Avatar`. All you need are the styles in the function below

Reply:
@anicholls What do you think of using Emotion's `as` to add and change some styles? https://emotion.sh/docs/styled#as-prop

It seems like we are extending `Avatar` to add additional styles and change the tag used. As this is currently implemented, we will have this DOM structure: `button > div > img|svg`. We don't need that extra `div`

Reply:
The `as` prop is [new in Emotion 10](https://medium.com/emotion-js/announcing-emotion-10-f1a4b17b8ccd#f8b7), so unfortunately we can't do this yet 😞

Reply:
ok but seems like 
```
padding: 0,
border: 0,
```
are needed too, will leave those as well

Reply:
Ah, yep - sorry forgot about the default button styling 👍

Reply:
Ah. I think the API is `withComponent` that would work in lower versions

Reply:
You have a point @NicholasBoll, the current structure for the rendered AvatarButton is now:
```
<button aria-label="Avatar">
  <div aria-label="Avatar">
    <img src="imageUrl" alt="Avatar">
  </div>
</button>
```
Which is not the best. 
Any recommendation here @anicholls?

Reply:
Yes, let's use [`withComponent`](https://emotion.sh/docs/styled#change-the-rendered-tag-using-withcomponent)